### PR TITLE
feat(create-medusa-app, medusa): check Node.js version before installation.

### PR DESCRIPTION
**Pull request by shahednasser**
Check before installation with `create-medusa-app` and `medusa new` whether the current node version is >= 20. If not, exit with an error.

### CODE REVIEW

File: packages/cli/create-medusa-app/src/commands/create.ts [hunk 1]
```diff
@@ -53,6 +53,13 @@ export default async ({
   withNextjsStarter = false,
   verbose = false,
 }: CreateOptions) => {
+  const nodeVersion = getNodeVersion()
+  if (nodeVersion < 20) {
+    logMessage({
+      message: `Medusa requires at least v20 of Node.js. You're using v${nodeVersion}. Please install at least v20 and try again: https://nodejs.org/en/download`,
+      type: "error"
+    })
```
Comment (riqwan) [inline packages/cli/create-medusa-app/src/commands/create.ts:61]: q: should we exit here as well?

**Review by riqwan — APPROVED** [review]
🥇

File: packages/core/utils/src/common/get-node-version.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export function getNodeVersion(): number {
+  const [major] = process.versions.node.split('.').map(Number)
+
+  return major
+}
```
Comment (riqwan) [inline packages/core/utils/src/common/get-node-version.ts]: super nit: 
```suggestion
}

export const MIN_SUPPORTED_NODE_VERSION = 20

```

**Review by olivermrbl — APPROVED** [review]
Other than @riqwan's comment, LGTM :)

File: packages/cli/create-medusa-app/src/commands/create.ts (hunk 1 above)
Comment (shahednasser) [reply packages/cli/create-medusa-app/src/commands/create.ts:61]: This one exits automatically when type is `error`.

**Pull request merged**